File: examples/sequential-java/Example0.java
/* This code adds all the values of an array */

public class Example0 {
	private int array[];
	private long result;
	
	public Example0(int array[]) {
		this.array = array;
		this.result = 0;
	}
	
	public long getResult() {
		return result;
	}
	
	public void calculate() {
		result = 0;
		for (int i = 0; i < array.length; i++) {
			result += array[i];
		}
	}
	
	public static void main(String args[]) {
		final int SIZE = 300_000_000;
		int array[] = new int[SIZE];
		long startTime, stopTime;
		double acum = 0;
		
		Utils.fillArray(array);
		Utils.displayArray("array", array);
		
		Example0 e = new Example0(array);
		acum = 0;
		for (int i = 0; i < Utils.N; i++) {
			startTime = System.currentTimeMillis();
			e.calculate();
			stopTime = System.currentTimeMillis();
			acum += (stopTime - startTime);
		}
		System.out.printf("sum = %d\n", e.getResult());
		System.out.printf("avg time = %.5f\n", (acum / Utils.N));
	}
}
			

File: examples/sequential-cpp/example0.cpp
/* This code calculates the sum of all elements in an array */
#include <iostream>
#include <cmath>
#include "utils/cppheader.h"

const int SIZE = 1000000000;

using namespace std;

class SumArray {
private:
	int *myArray, mySize;
	double result;
	
public:
	SumArray(int *array, int size) : myArray(array), mySize(size), result(0) {}
	
	double getResult() const {
		return result;
	}
	
	void calculate() {
		result = 0;
		for (int i = 0; i < mySize; i++) {
			result += myArray[i];
		}
	}
};

int main(int argc, char* argv[]) {
	double ms;
	Timer t;
	int *a;
	
	a = new int[SIZE];
	fill_array(a, SIZE);
	display_array("a", a);
	
	
	ms = 0;
	SumArray obj(a, SIZE);
	cout << "Starting..." << endl;
	for (int i = 0; i < N; i++) {
		t.start();
		obj.calculate();
		ms += t.stop();
	}
	cout << "sum = " << obj.getResult() << endl;
	cout << "avg time = " << (ms / N) << " ms" << endl;
	
	delete [] a;
	return 0;
}
